scraper/src/mindtouch2zim/libretexts/glossary.py

```diff
@@ -0,0 +1,62 @@
+from bs4 import BeautifulSoup
+
+
+class GlossaryRewriteError(Exception):
+    """Exception indicating a problem during glossary rewrite"""
+
+    pass
+
+
+def _get_formatted_glossary_row(row) -> str:
+    """Format one row as HTML"""
+    word = row.find("td", attrs={"data-th": "Word(s)"}).text
+    definition = row.find("td", attrs={"data-th": "Definition"}).text
+    return (
+        '<p class="glossaryElement">\n'
+        f'  <span class="glossaryTerm">{word}</span>\n'
+        "  |\n"
+        f'  <span class="glossaryDefinition">{definition}</span>\n'
+        "</p>\n"
+    )
+
+
+def rewrite_glossary(original_content: str) -> str:
+    """Statically rewrite the glossary of libretexts.org
+
+    Only word and description columns are supported.
+    """
+
+    soup = BeautifulSoup(
+        original_content,
+        "html.parser",  # prefer html.parser to not add <html><body> tags
+    )
+
+    glossary_table = None
+
+    for table in soup.find_all("table"):
+        if not table.caption:
+            continue
+        if table.caption and table.caption.text.strip() == "Example and Directions":
+            continue
+        if glossary_table:
+            raise GlossaryRewriteError("Too many glossary tables")
+        glossary_table = table
+
+    if not glossary_table:
+        raise GlossaryRewriteError("Glossary table not found")
+
+    tbody = glossary_table.find("tbody")
+    if not tbody:
+        raise GlossaryRewriteError("Glossary table body not found")
```

Comment:
I think it works fine for now, given its rudimentary, but i think that using selectors that beautiful soup supports would be more appropriate in general: you check for the exact thing you want instead of looking and failing on missing into the whole tree

Reply:
This was done on-purpose to give more precise error message ... but I agree it comes at the expense of more code to read / write ...